x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichPolicyRunner.java

```diff
@@ -196,29 +196,29 @@ private void validateField(Map<?, ?> properties, String fieldName, boolean field
 
     private XContentBuilder resolveEnrichMapping(final EnrichPolicy policy) {
         // Currently the only supported policy type is EnrichPolicy.MATCH_TYPE, which is a keyword type
-        String keyType;
+        final String keyType;
+        final MatchFieldMapping matchFieldMapping;
         if (EnrichPolicy.MATCH_TYPE.equals(policy.getType())) {
-            keyType = "keyword";
+            matchFieldMapping = (builder) -> builder.field("type", "keyword").field("doc_values", false);
             // No need to also configure index_options, because keyword type defaults to 'docs'.
+        } else if (EnrichPolicy.GEO_MATCH_TYPE.equals(policy.getType())) {
+            matchFieldMapping = (builder) -> builder.field("type", "geo_shape");
```

Comment:
I don't believe that `geo_shape` field type uses doc values by default, right?
Do you know whether the default is going to change in the future?
I just want to make sure we never store more than what is needed for enrich. (since doc values are not being used for querying)

Reply:
the problem is that since doc-values aren't supported, neither is `doc-values: false`. I was debating whether this should be fixed upstream so that one is allowed to set that.

Reply:
I will follow-up in another PR to expose `doc-values` parsing to `geo_shape`. I think it should understand that parameter, even though it only allows it to be `false`. For now, I think this is the best that can be done. Thankfully, I know when doc-values on shapes will be supported so I will be sure to update this!

Reply:
👍 That sounds good to me.